common-rendering/src/components/keyEvents.tsx

```diff
@@ -0,0 +1,187 @@
+import { palette, remSpace } from "@guardian/src-foundations";
+import { SvgChevronUpSingle, SvgChevronDownSingle } from "@guardian/src-icons";
+import { Pillar, Theme } from "@guardian/types";
+import { from } from "@guardian/src-foundations/mq";
+
+// ----- Component ----- //
+type paletteId = 300 | 400;
+
+interface KeyEvent {
+	time: string;
+	text: string;
+	url: string;
+}
+
+interface KeyEventsProps {
+	keyEvents: KeyEvent[];
+	theme: Theme;
+}
+
+interface ListItemProps {
+	keyEvent: KeyEvent;
+	theme: Theme;
+}
+
+const getColor = (theme: Theme, paletteId: paletteId) => {
+	switch (theme) {
+		case Pillar.Sport:
+			return palette.sport[paletteId];
+		case Pillar.Culture:
+			return palette.culture[paletteId];
+		case Pillar.Lifestyle:
+			return palette.lifestyle[paletteId];
+		case Pillar.Opinion:
+			return palette.opinion[paletteId];
+		default:
+			return palette.news[paletteId];
+	}
+};
+
+const keyEventWrapperStyles: SerializedStyles = css`
+	width: 100%;
+
+	${from.desktop} {
+		width: 13.75rem;
+	}
+`;
+
+const detailsStyles: SerializedStyles = css`
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const titleRowStyles: SerializedStyles = css`
+	position: relative;
+	margin: ${remSpace[1]} ${remSpace[1]} 0 ${remSpace[3]};
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+
+	&:focus {
+		outline: none;
+	}
+
+	path {
+		fill: ${palette.neutral[46]};
+	}
+
+	svg {
+		height: 2rem;
+	}
+
+	${from.phablet} {
+		margin: ${remSpace[1]} ${remSpace[4]} 0;
+	}
+
+	${from.desktop} {
+		display: none;
+	}
+`;
+
+const titleStyle: SerializedStyles = css`
+	${headline.xxsmall({ fontWeight: "bold", lineHeight: "tight" })};
+	color: ${palette.neutral[7]};
+`;
+
+const listStyles: SerializedStyles = css`
+	margin: ${remSpace[3]};
+
+	${from.phablet} {
+		margin: ${remSpace[3]} ${remSpace[5]} 0;
+	}
+
+	${from.desktop} {
+		margin: ${remSpace[1]} 0 0;
+	}
+
+	li::before {
+		content: "";
+		border-color: transparent ${palette.neutral[7]};
+		border-style: solid;
+		border-width: 0.4rem 0 0.4rem 0.5rem;
+		display: block;
+		height: 0;
+		width: 0;
+		top: 0;
+		position: absolute;
+	}
+`;
+
+const listItemStyles: SerializedStyles = css`
+	padding-bottom: ${remSpace[3]};
+	border-left: 1px solid ${palette.neutral[7]};
+	position: relative;
+
+	&:last-child {
+		border-left: none;
+	}
+`;
+
+const timeTextWrapperStyles: SerializedStyles = css`
+	margin-left: ${remSpace[4]};
+`;
+
+const textStyles = (theme: Theme): SerializedStyles => css`
+	${headline.xxxsmall({ fontWeight: "regular", lineHeight: "regular" })};
+	color: ${getColor(theme, 300)};
+	text-decoration: none;
+`;
+
+const timeStyles: SerializedStyles = css`
+	${textSans.xxsmall({ fontWeight: "bold", lineHeight: "tight" })};
+	color: ${palette.neutral[7]};
+	display: block;
+`;
+
+const ListItem: FC<ListItemProps> = ({ keyEvent, theme }) => {
```

Comment:
I realise this is subjective but the problem with using `React.FC` to type components is you're actually adding a `children` prop to this component, which isn't being used and is kind of weird. I personally prefer to be more explicit when typing things.

More opinionated [ramblings on this here](https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript).

Reply:
Interesting point 🤔. What would you use instead?

Reply:
A react component is just a javascript function so I don't think you need to do anything special with it. Most of the DCR components are simply:

```
const MyThing = ({foo, bar}: Props) => { return (<p>My Thing</p>)}; 
```

Reply:
Or even

```
function MyThing({foo, bar}: Props) { return (<p>My Thing</p>)}; 
```

Equally valid!

Reply:
So my understanding here is that there are, typically, three things you can do with the return type for a React component:

#### Use `React.FC`
This is very wide. It always adds `children` which is usually wrong.

#### Specify your own type
This is just hard. You can pick from `JSX.Element`, `React.ReactNode`, `string`, `JSX.Element[]`, `null`, `undefined`, `React.ReactElement`, etc. or combinations of them but there's no one solution fits all and it's very easy to get this wrong triggering errors later on for people consuming your component.

#### Nothing, and let Typescript imply the type
In most cases, this works. Typescript is clever enough to know what the type should be using inference and in those situations where the component is more complex, you can choose to specify a type.

I, personally, like the last option - doing nothing - because it's not only easier but it means that when a type is specified it shows that the developer really wants to communicate something that I should pay attention to. But I also don't think it's that big of a deal or in any way a blocker for this PR - it's basically just a fun, interesting discussion.